### PR TITLE
Tempo: Add query ref in the query editor

### LINKED ISSUE #81192: Tempo Explore: Under certain circumstances the "disable" button is toggled on query edit
The disable query button will somtimes toggle on/off simply by editing a query:

image

Then by simply adding a space to the query the query is re-enabled. I have gotten this to toggle the other way as well. Going from enabled -> disabled on edit.

image

This may not be directly related to "Copy Query From Search" but I can reliably repro the issue with that feature.

Steps to repro:

1. Load Explore/Tempo Datasource
2. Go to the "Search" tab
3. Select a service name from the drop down
4. Go to the "TraceQL" tab
5. Click "Copy query from search"
6. Click the disable query/eyeball button. (The query is now disabled)
7. Edit the query in any way. (The query is toggled back to enabled).

### LINKED ISSUE #81193: Tempo Explore: Editing a TraceQL query toggles the "Table format"
Any edit to a TraceQL query will remove the preferred format:

image

Steps to repro:

1. Go to explore
2. Add a simple TraceQL query like `{}`
3. Select "Spans"
4. Click "Run Query"
5. Make any edit to the query
6. Notice that the table format toggles back to "Traces"

### PR DESCRIPTION
**Pull request by joey-grafana**
**What is this feature?**

Adds a query ref in the Tempo query editor.

**Why do we need this feature?**

We pass in the query to the Monaco editor but do not also pass updates to the query. So, when the query changes and then the text in the Monaco field (i.e. the TraceQL field) is updated, Monaco will update the Tempo query with the new TraceQL query but unfortunately uses the original Tempo query (without any updates from the user) it was given in the Query Editor `onEditorChange` and so any changes will not be reflected. 

In this PR we add a ref so that Monaco always has the latest values for query.

**Who is this feature for?**

Tempo users

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/81192
Fixes https://github.com/grafana/grafana/issues/81193

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by fabrizio-grafana — APPROVED** [review]
> Fixes
> https://github.com/grafana/grafana/issues/81192
> https://github.com/grafana/grafana/issues/81193

I'm not sure if those fixes will be closed automatically if reported like that. In fact if I open them, I don't see this PR referenced there. I _think_  the correct way is:

> Fixes https://github.com/grafana/grafana/issues/81192
> Fixes https://github.com/grafana/grafana/issues/81193

but again, not sure, feel free to ignore the comment!

File: public/app/plugins/datasource/tempo/traceql/QueryEditor.tsx [hunk 1]
```diff
@@ -27,6 +27,10 @@ export function QueryEditor(props: Props) {
     return genQuery === query.query || genQuery === '{}';
   });
 
+  // The Monaco Editor uses the first version of onChange in handleOnMount i.e. always has the initial
```
Comment (fabrizio-grafana) [inline public/app/plugins/datasource/tempo/traceql/QueryEditor.tsx:30]:
```suggestion
  // The Monaco Editor uses the first version of props.onChange in handleOnMount i.e. always has the initial
```

**Review by fabrizio-grafana — APPROVED** [review]

**Review by fabrizio-grafana — APPROVED** [review]

**Comment by aocenas** [comment]
@joey-grafana I don't think this is best way to fix this. 

1. we could and probably should fix it in the CodeEditor itself. This is not even documented but is important caveat to the API. We should either create an issue and let CodeEditor owners handle it or propose a fix.
2. Even if we weren't going to fix this, the issue is in CodeEditor which is wrapped in TraceQLEditor but we are fixing it in QueryEditor. This is 2 levels separated from the root cause. So anybody who uses TraceQLEditor in some other component would have to first know this is an issue (as this is mentioned nowhere in the TraceQLEditor it's doubtful they will know) and will also need a copy of this same fix. Moving it down a level into `TraceQLEditor` would at least isolate this issue more.

**Comment by joey-grafana** [comment]
@aocenas yeah sounds good!
1. https://github.com/grafana/grafana/issues/81687
2. https://github.com/grafana/grafana/pull/81686

**Pull request merged**